go/tools/gazelle/rules/sort_labels.go

```diff
@@ -0,0 +1,128 @@
+package rules
+
+import (
+	"sort"
+	"strings"
+
+	bf "github.com/bazelbuild/buildtools/build"
+)
+
+var (
+	goRuleKinds = map[string]bool{
+		"cgo_library": true,
+		"go_binary":   true,
+		"go_library":  true,
+		"go_test":     true,
+	}
+	sortedAttrs = []string{"srcs", "deps"}
+)
+
+// SortLabels sorts lists of strings in "srcs" and "deps" attributes of
+// Go rules using the same order as buildifier. Buildifier also sorts string
+// lists, but not those involved with "select" expressions.
+// TODO(jayconrod): remove this when bazelbuild/buildtools#122 is fixed.
+func SortLabels(f *bf.File) {
+	for _, s := range f.Stmt {
+		c, ok := s.(*bf.CallExpr)
+		if !ok {
+			continue
+		}
+		r := bf.Rule{c}
+		if !goRuleKinds[r.Kind()] {
+			continue
+		}
+		for _, key := range []string{"srcs", "deps"} {
+			attr := r.AttrDefn(key)
+			if attr == nil {
+				continue
+			}
+			bf.Walk(attr.Y, func(e bf.Expr, _ []bf.Expr) {
+				list, ok := e.(*bf.ListExpr)
```

Comment:
I would probably have extracted this to a named function.
I don't think it's closing over any variables, so it's reasonably discrete, and I generally try to avoid long lambdas with return statements, it's really hard to see what the return will do.